TheMET/TheMET/ViewControllers/FavoritesViewController.swift

```diff
@@ -37,43 +42,74 @@ class FavoritesViewController: UIViewController, UITableViewDelegate, UITableVie
+           medium.isEmpty == false {
+            cellText.append(medium + "\n")
+        }
+        return cellText
+    }
+    
+    private func loadCellTags(cell: ArtViewCell, indexPath: IndexPath) -> [String] {
+        var tags: [String] = []
+        if let departmentName = self.favoriteService.favoriteArts[indexPath.row].department {
+            tags.append(departmentName)
+        }
+        return tags
+    }
+    
+    private func reloadFavoritesViewController() {
         
     }
     
 //  UITableViewDelegate, UITableViewDataSource
     
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         if let cell = tableView.dequeueReusableCell(withIdentifier: ArtViewCell.artViewCellIdentifier, for: indexPath) as? ArtViewCell {
-            if indexPath.row == 0 {
-                cell.backgroundColor = .clear
-                cell.selectionStyle = .none
-                cell.imageState = .loaded(UIImage(named: "Image1")!)
-                cell.isLiked = true
-                cell.text = "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam"
-                cell.onLikeButtonDidTap = {
-                    print("Test art with id 0 favourite button tapped")
-                }
-                cell.tags = ["aaaaaaa", "bbbbbbb", "ccccccc", "dddddddd", "eeeeeeeee"]
-            } else if indexPath.row == 1 {
-                cell.backgroundColor = .clear
-                cell.selectionStyle = .none
-                cell.imageState = .loaded(UIImage(named: "Image2")!)
-                cell.isLiked = true
-                cell.text = "quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum "
-                cell.onLikeButtonDidTap = {
-                    print("Test art with id 1 favourite button tapped")
-                }
-                cell.tags = ["aaaaaaa", "bbbbbbb", "ccccccc", "dddddddd", "eeeeeeeee"]
-            } else {
-                cell.backgroundColor = .clear
-                cell.selectionStyle = .none
-                cell.imageState = .loaded(UIImage(named: "Not Loaded Image")!)
-                cell.isLiked = true
-                cell.text = "quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum "
-                cell.onLikeButtonDidTap = {
-                    print("Test art with id 2 favourite button tapped")
-                }
-                cell.tags = ["aaaaaaa", "bbbbbbb", "ccccccc", "dddddddd", "eeeeeeeee"]
+            cell.backgroundColor = .clear
+            cell.selectionStyle = .none
+            self.loadCellImage(cell: cell, indexPath: indexPath)
+            cell.isLiked = true
+            cell.text = self.loadCellText(cell: cell, indexPath: indexPath)
```

Comment:
I guess you can extract and `art` from the `FavoriteService` only once and then pass it to your methods instead of `indexPath`

Reply:
+